### ISSUE TITLE
ui-sref in index.html

### ISSUE THREAD

**Issue by ghost** [ColinTheRobot/ang-auth#14]
Can someone please explain what ui-sref refers too? Its assigined to the buttons in index.html. Does it have to do with the router?

**Comment by ztretterath** [comment]
Check out router.js

all the different states

then look at the ui-sref

:)

**Comment by estermer** [comment]
it is like declaring a href but links it to the ui-router instead of a simple path

**Comment by mraballard** [comment]
yes it refers to the ui-router states in router.js ...  it's like href="", but instead of giving it a URL page that will be used to reload the page, it just loads the state from router.js without reloading the page

**Comment by ghost** [comment]
Thanks guys !

**Comment by estermer** [comment]
this of the 's' in the ui-sref as standin for 'state' it is 'referencing' where the 'h' in 'href' stands for 'hyperlink' that is 'referencing'

**Comment by RBeneroff** [comment]
Okay, I understand that using ui-sref is in place of href so it brings you do the edit page when you click Edit, but why is {todo: todo} being passed in instead of just todo?

**Comment by estermer** [comment]
@RBeneroff where do you mean that is being passed Rachel?

**Comment by RBeneroff** [comment]
@estermer 
<button ui-sref="editTodo({todo: todo})" 
why is it {todo: todo} instead of just editTodo(todo)

I know this is probably a simple answer, but right now I am just not seeing it.

**Comment by estermer** [comment]
`state('editTodo', {
      url: '/todos/edit',
      params: {
        todo: null //<---- because todo is being passed into the todos parameter
      }`

**Comment by RBeneroff** [comment]
ahhh, thank you! @estermer

**Comment by estermer** [comment]
yeah no problem

**Comment by ghost** [comment]
I have a question about this as well... ui-sref="editTodo({todo: todo})"  the first todo: has to do with? 
the second : todo has to do with?  This is where I get confused and just start naming everything the same so I don't get it wrong ha!

**Comment by tillydray** [comment]
@jesseklein14 `{todo: todo}` is `{key: value}`, where `key` is the name we will use later to refer to it, and `value` is what the key is equal to.  Think of it like `{name: 'Jesse'}`.  

`ui-sref="editTodo({todo: todo})` is passing an object (or probably json, not sure) as an argument to editTodo, where the key is `todo` and the value is `todo`.  The value `todo` includes the information that the user just edited, and will be used in app.js line 49 `$http.put(`/todos/${$state.params.todo._id}`, todo)`.  Line 49 is adding hitting todos.js line 62, the put route, which is updating the correct todo by checking it's id.

**Comment by ghost** [comment]
@Auslegung ---  thank you!
